### PR TITLE
Feature/serde

### PR DESCRIPTION
**Pull request by hoodie**
how about we add some serde support while we're at it?

this only contains the deserializer implementation, and I need to do some homework on how to write unit tests without pulling in an extra dependency, but what do you think?

(follow up to #48)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,8 +19,9 @@ edition = "2018"
 nom = { version = "7", default-features = false }
 chrono = { version = "0.4", default-features = false, optional = true }
 num-traits = { version = "0.2", optional = true }
+serde = { version = "1.0", optional = true }
 
 [features]
-default = ["std"]
+default = ["std", "serde"]
```
Comment (badboy) [inline Cargo.toml:25]: Can we keep that out of the defaults for now?

File: src/serde.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        deserializer.deserialize_str(DateVisitor)
+    }
+}
+
+struct DateVisitor;
+impl<'de> Visitor<'de> for DateVisitor {
+    type Value = crate::Date;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a string that follows iso8691 Date format")
```
Comment (badboy) [inline src/serde.rs:23]:
```suggestion
        write!(formatter, "a string that follows ISO8601 date format")
```

8691? what are you trying to set on fire?

File: src/serde.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        deserializer.deserialize_str(TimeVisitor)
+    }
+}
+
+struct TimeVisitor;
+impl<'de> Visitor<'de> for TimeVisitor {
+    type Value = crate::Time;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a string that follows iso8691 Time format")
```
Comment (badboy) [inline src/serde.rs:51]:
```suggestion
        write!(formatter, "a string that follows ISO8601 time format")
```

**Review by badboy — APPROVED** [review]
nits, but if you address them good to go.

File: src/serde.rs (hunk 2 above)
Comment (hoodie) [reply src/serde.rs:23]: OMG, LOL, that's why I don't merge my late night commits myself 🤣

File: Cargo.toml (hunk 1 above)
Comment (hoodie) [reply Cargo.toml:25]: * [ ] is there a way to make this a merge blocker?

Comment (badboy) [reply Cargo.toml:25]: What do you mean?

Comment (hoodie) [reply Cargo.toml]: so that I have to check some box before I can merge

**Pull request merged**